File: kotlin/src/main/kotlin/introduction/RandomWalk.kt
package introduction

import processing.core.PApplet
import processing.core.PConstants

class RandomWalk : PApplet() {
    var x: Float = 0.0f
    var y: Float = 0.0f

    override fun settings() {
        size(700, 500, PConstants.P2D)
    }

    override fun setup() {
        background(255)
        x = width * 0.5f
        y = height * 0.5f
    }

    override fun draw() {
        x += random(-1.0f, 1.0f)
        y += random(-1.0f, 1.0f)

        point(x, y)
    }
}

private fun main() {
    PApplet.main("introduction.RandomWalk")
}

File: p5js/introduction/random_walk.js
const walker = { x: 0.0, y: 0.0 };

function setup() {
    createCanvas(640, 240);
    background(255);

    walker.x = 320;
    walker.y = 120;
}

function step() {
    let choice = parseInt(random(5));
    switch (choice) {
        case 0:
            walker.x++;
            break;

        case 1:
            walker.x--;
            break;

        case 2:
            walker.y++;
            break;
        
        default:
            walker.y--;
            break;
    }
}

function draw() {
    step();

    stroke(0);
    point(walker.x, walker.y);
}